src/components/views/elements/MemberEventListSummary.js

```diff
@@ -144,87 +144,105 @@ module.exports = React.createClass({
         );
     },
 
-    render: function() {
-        let summary = null;
+    shouldComponentUpdate: function(nextProps, nextState) {
+        return (
+            nextProps.events.length !== this.props.events.length ||
```

Comment:
> @Kegsay said something along the lines of if the parent gets updated, then it will re-render each child component, but only if shouldComponentUpdate is true

I don't think that's correct. I believe it will only re-render children which are returned in the virtual DOM: if the summary is not expanded, the child components are not returned by `MemberEventListSummary.render`, so their `render` methods are not called.

> So as an example, it won't now re-render the MemberEventListSummary if a message is added to the scroll panel.

yup... but I'm not sure if that's a worthwhile saving, having fixed render.